magento/magento2
# Changed file permissions from 755 to 644 of non-executable files in git.

### Description (*)
This was triggered by https://github.com/magento/magento2/commit/90df53c49a8f1cdd70289b7b33523563c3074197#commitcomment-34538350

### Fixed Issues (if relevant)
1. https://github.com/magento/magento2/issues/1453: Permissions not set correctly
(it was already fixed before, but until there is some automated test pointing out these problems, it will keep happening I'm afraid)

This corrects the file permissions for the following file types from 755 to 644:
```
.bmp
.css
.eot
.gif
.gitignore
.htaccess
.ico
.jpg
.js
.json
.less
.map
.md
.php
.phtml
.png
.svg
.ttf
.txt
.woff
.woff2
.xml
.xsd
```

### Manual testing scenarios (*)
1. Clone the M2 repo
2. Execute: `find . -type f ! -path './.git/*' ! -perm 644`
3. This should only report executable files, at the time of writing those are:
```
./bin/magento
./dev/tests/integration/bin/magento
./dev/travis/before_install.sh
./dev/travis/before_script.sh
```

### Questions or comments
In order to prevent this from happening in the future, it would be nice if some sort of test would get created which can detect this.
A potential test for this could be:
1. Execute `find . -type f ! -path './.git/*' ! -perm 644`
2. For all the files it finds, if one of those does not start with `#!/usr/bin/env` on the first line, it should get flagged.

### Contribution checklist (*)
